### PR TITLE
Deploy server hookup

### PR DESCRIPTION
**Pull request by rafkhan**
Connected to #307

### CODE REVIEW

File: src/api/0.1/js/js-api.js [hunk 1]
```diff
@@ -182,12 +181,14 @@ function deploymentExists(projectId) {
...
-  const pm = getProjectAPI();
-  return deploy_(pm, projectId, deploymentDesc, name, sshData, force)
-    .fail((err) => {
-      util.info('Clusternator: Error creating deployment: ' + err.message);
-      util.info(err.stack);
-    });
+  return getProjectAPI()
+    .createDeployment(projectId, name, deploymentDesc, sshData, force);
+
+  //return deploy_(pm, projectId, deploymentDesc, name, sshData, force)
```
Comment (SethDavenport) [inline src/api/0.1/js/js-api.js]: Commented out code!

File: src/clusternator/projectManager.js [hunk 2]
```diff
@@ -54,16 +58,30 @@ function createPR() {
...
 
+  return deploymentsREST.create(projectId, name, deploymentDesc, sshData, force)
+    .then((response) => {
+      util.info('Successfully deployed ' + fullName, response);
+    }, (err) => {
+      util.error('Failed to deploy ' + fullName, err.stack);
+    });
 }
 
 function destroy() {
```
Comment (SethDavenport) [inline src/clusternator/projectManager.js]: What's this for?

File: src/clusternator/projectManager.js [hunk 3]
```diff
@@ -54,16 +58,30 @@ function createPR() {
 
 }
 
-function createDeployment() {
+function createDeployment(projectId, name, deploymentDesc, sshData, force) {
+  let fullName = projectId + ':' + name;
 
+  return deploymentsREST.create(projectId, name, deploymentDesc, sshData, force)
+    .then((response) => {
+      util.info('Successfully deployed ' + fullName, response);
```
Comment (bennett000) [inline src/clusternator/projectManager.js]: @rafkhan these console logs should flow all the way out to the CLI endpoints.  Let them handle the pass/fail.  That way if another library uses clusternator as a library their stdout won't be polluted.

File: src/clusternator/projectManager.js [hunk 4]
```diff
@@ -54,16 +58,30 @@ function createPR() {
...
+    }, (err) => {
+      util.error('Failed to deploy ' + fullName, err.stack);
```
Comment (bennett000) [inline src/clusternator/projectManager.js]: flow out to cli

File: src/clusternator/projectManager.js [hunk 5]
```diff
@@ -54,16 +58,30 @@ function createPR() {
...
 
 }
 
-function destroyDeployment() {
+function destroyDeployment(projectId, name) {
+  let fullName = projectId + ':' + name;
 
+  return deploymentsREST.destroy(projectId, name)
+    .then((response) => {
+      util.info('Successfully destroyed' + fullName, response);
```
Comment (bennett000) [inline src/clusternator/projectManager.js]: flow out...

File: src/clusternator/projectManager.js [hunk 6]
```diff
@@ -54,16 +58,30 @@ function createPR() {
...
+    }, (err) => {
+      util.error('Failed to destroy' + fullName, err.stack);
```
Comment (bennett000) [inline src/clusternator/projectManager.js]: flow out...

File: src/aws/projectManager.js [hunk 7]
```diff
@@ -228,8 +228,9 @@ function getProjectManager(ec2, ecs, awsRoute53, dynamoDB, awsIam, awsEcr,
   function destroyDeployment(projectId, dep) {
     return state()
       .then((s) => findOrCreateProject(projectId)
-        .then((snDesc) => s
-          .deployment.destroy( projectId, dep)));
+        .then((snDesc) => {
+          return s.deployment.destroy(projectId, dep);
+        }));
```
Comment (bennett000) [inline src/aws/projectManager.js]: This change doesn't actually do anything :)

**Comment by bennett000** [comment]
Oh btw :shipit: when the logs are relocated.

File: src/api/0.1/js/js-api.js [hunk 8]
```diff
@@ -113,7 +113,6 @@ function login(username, password) {
 
 /**
  * @param {string} username
- * @param {string} password
```
Comment (bennett000) [inline src/api/0.1/js/js-api.js]: How come this line got deleted, looks like the parameter is still there :/

**Comment by bennett000** [comment]
@rafkhan this is looking good with the exception of the one deleted comment

File: src/api/0.1/js/js-api.js (hunk 8 above)
Comment (rafkhan) [inline src/api/0.1/js/js-api.js]: wow idk, I must have missed it in the diff

**Comment by bennett000** [comment]
:shipit:

**Pull request merged**